Kernel/Arch/riscv64/CPU.cpp

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright (c) 2024, Leon Albrecht <leon.a@serenityos.org>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include <Kernel/Arch/riscv64/CPU.h>
+#include <Kernel/Memory/MemoryManager.h>
+#include <Userland/Libraries/LibDeviceTree/FlattenedDeviceTree.h>
+#include <Userland/Libraries/LibDeviceTree/Validation.h>
+
+namespace Kernel {
+
+BootInfo s_boot_info;
+
+void dump_fdt(PhysicalPtr fdt_phys_ptr)
```

Comment:
Do the two VM regions get automatically dropped on function exit? Otherwise we should clear them up even though the physical memory is unaffected.

Reply:
The regions are in own pointers, so when they go out of scope the regions are destroyed, and that unregisters them from the memory map to my knowledge